enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+---
+title: assisted-installer-in-cluster
+authors:
+  - "@mhrivnak"
+  - "@hardys"
+  - "@avishayt"
+reviewers:
+  - "@eparis"
+  - "@markmc"
+  - "@dgoodwin"
+  - "@ronniel1"
+
+approvers:
+  - TBD
+creation-date: 2020-12-22
+last-updated: 2020-12-22
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced|informational
+see-also:
+replaces:
+superseded-by:
+---
+
+# Assisted Installer in-cluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Assisted Installer currently runs as a SaaS on cloud.redhat.com, enabling
+users to deploy OpenShift clusters with certain customizations, particularly on
+bare metal hardware. It is necessary to bring those capabilities on-premise in
+users' clusters for two purposes:
+
+1. A cluster created by the Assisted Installer needs the ability to add workers
+on day 2.
+2. Multi-cluster management, such as through hive and ACM, should be
+able to utilize the capabilities of the Assisted Installer.
+
+This enhancement proposes the first iteration of running the Assisted Installer
+in end-user clusters to meet the purposes above.
+
+## Motivation
+
+### Goals
+
+* Expose the Assisted Installer's capabilities as a kubernetes-native API.
+* Enable any OpenShift cluster deployed by the Assisted Installer to add worker nodes on day 2.
+* Enable multi-cluster management tooling to create new clusters using Assisted Installer.
+* Utilize [metal3](https://metal3.io/) to boot the assisted discovery ISO for bare-metal deployments.
+* Ensure the design is be extensible for installing on other platforms.
+* Utilize the Assisted Service and Agent with minimal changes to avoid
```

Comment:
I don't think it's a good idea to treat this as an absolute constraint like this. It too easily turns into a blanket excuse for any time we start to feel like the best design is too hard.

For sure, mention it as a consideration we should bear in mind.

Reply:
I'll move this under "implementation details/notes/constraints" and try to make it read as less of an absolute.